src/utils.c

```diff
@@ -221,6 +222,73 @@ data_left (int soc)
   return data;
 }
 
+/**
+ * @brief Check if the hosts is still alive and set it as dead if not.
+ *
+ * @param kb Host kb where the host is set as dead.
+ *
+ * @return 1 if considered alive, 0 if it is dead. -1 on error
+ * or option disabled.
+ */
+int
+check_host_still_alive (kb_t kb, const char *hostname)
```

Comment:
As suggested above. Maybe move counter logic into other function. Also rename variables according to settings name change (something like heartbeat -> max_nvt_timouts).